src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/GetCaseResponse.java

```diff
@@ -62,8 +64,8 @@ public class GetCaseResponse {
     @JsonProperty("stages")
     private List<SimpleStageDto> stages;
 
-    @JsonProperty("completed")
-    private Boolean completed;
+    @JsonProperty(value="completed", access=READ_ONLY)
+    private LocalDateTime dateCompleted;
```

Comment:
This is still not providing backwards compatibility for consumers of the existing api. Previously making a request to  `/case/data/{reference}` would have responded with an object including: 
```json
{
  // ... more fields
  "completed": true,
  // ... more fields
}
```
but now it will return something like:

```json
{
  // ... more fields
  "completed": "2023-09-12T01:02:03.456Z",
  // ... more fields
}
```

This will cause errors in consumers of the api, so we still need to provide completed as a boolean to existing API clients that expect it to be there. This ticket is concerned about there being a single source of truth in the DB for if a case is completed or not, not about also updating each service to switch to the new field right away.


```suggestion
    @JsonProperty("dateCompleted")
    private LocalDateTime dateCompleted;
    
    @JsonProperty(value="completed", access=READ_ONLY)
    public Boolean getCompleted() {
        return dateCompleted != null;
    }
```

Reply:
I have incorporated above review comments and also added test cases to depend on dateCompleted.